### PR TITLE
Clarify the Conventions in Pulser

### LINKED ISSUE #571: Write a "Conventions" page
Write a "Conventions" page to add under the "Fundamental Concepts" section of the Pulser docs. It should mention:

- [x] The different Hamiltonians and respective bases  
- [x] The single-qutrit state and the projections into the single-qubit states in the different bases 
- [x] Multi-partite state order
- [x] Measurement convention: what's 0 and what's 1 (depending on basis), order in a measurement sample

### PR DESCRIPTION
**Pull request by HGSilveri**
- Adds a "Conventions" page to the docs, establishing all the conventions in Pulser
- Modifies the simulation in XY mode to comply with this convention

Closes #571 .

### CODE REVIEW

File: tutorials/quantum_simulation/Spin chain of 3 atoms in XY mode.ipynb
Comment (a-corni) [inline tutorials/quantum_simulation/Spin chain of 3 atoms in XY mode.ipynb:20]: Won't it be clearer to have the units not as a fraction ?

File: docs/source/conventions.rst
Comment (a-corni) [inline docs/source/conventions.rst:10]: Nit: Just wondering if the "Essentially" is needed "A basis refers to a set of two eigenstates. The transition between" or "A basis essentially refers to a set of two eigenstates. The transition between"

**Review by a-corni — COMMENTED** [review]
Great documentation, as far as I am concerned it is very clear and it will for sure help users :) Just found a minor typo and have two suggestions

File: tutorials/quantum_simulation/Spin chain of 3 atoms in XY mode.ipynb
Comment (HGSilveri) [reply tutorials/quantum_simulation/Spin chain of 3 atoms in XY mode.ipynb:20]: To me it doesn't make a big difference, I'll change it

File: docs/source/conventions.rst
Comment (HGSilveri) [reply docs/source/conventions.rst:10]: I agree with you, I'll remove it

File: pulser-simulation/pulser_simulation/simulation.py
Comment (a-corni) [inline pulser-simulation/pulser_simulation/simulation.py:405]: Thinking about it, I find it weird to have |0> -> "u" (for up ?) and |1> -> "d" (for down I guess) in the code. Won't it make more sense to invert the order of the "u" and "d" when building self.basis in _build_basis_and_op_matrices ?

**Review by a-corni — APPROVED** [review]

**Comment by madagra** [comment]
@HGSilveri I have only a couple of small comments but it all looks good to me:

* I would add a note about the endianness to which the chosen basis state ordering corresponds when expressing bitstrings. I think it is little endian, if I am not mistaken. The "Measurements samples ordering" is probably the right place for it.
* In the XY basis, however, the state ordering looks flipped to me, now you get big endian when you sample. I guess the explanation given to @a-corni is valid also in this case but I want to point out this since it can be pretty dangerous in some cases.

**Review by madagra — APPROVED** [review]

**Comment by HGSilveri** [comment]
> @HGSilveri I have only a couple of small comments but it all looks good to me:
> 
> * I would add a note about the endianness to which the chosen basis state ordering corresponds when expressing bitstrings. I think it is little endian, if I am not mistaken. The "Measurements samples ordering" is probably the right place for it.
> * In the XY basis, however, the state ordering looks flipped to me, now you get big endian when you sample. I guess the explanation given to @a-corni is valid also in this case but I want to point out this since it can be pretty dangerous in some cases.

Thanks for the review @madagra ! I'm not sure I understand your comments. From my understanding, endianness refers to how a binary number is stored in a register and displayed, so it is only relevant when you are reading your samples as a binary number. I think it's preferable to leave this up to the user and stick solely to matching the samples to the order of the qubit IDs. Let me give you an example:

Take a register with qubit IDs `("fish", "cat", "dog")`. The state vector is going to be `|fish, cat, dog>` and let's say that upon measurement in the ground-rydberg basis it is projected onto `|rgg>`. Then, we will register a count in sample `100`. That's all. The user can then decide whether "dog" or "fish" is the smallest-memory site in their Register and this will determine whether they read the sample as "4" or "1".

Also, can you point me to where ordering is flipped in the XY basis? This shouldn't be the case.

**Comment by lvignoli** [comment]
I had a quick read without checking the validity of the maths.
From a user that comes back to these computations regularly, I would say most of the useful information is there. Nice job!

File: docs/source/conventions.rst
Comment (lvignoli) [inline docs/source/conventions.rst:140]: I would put in the body text that b is the state of higher energy, not only in the warning below.
Also, a figure of the energy levels annotated with Ω, δ etc would make everything super clear.

```suggestion
The driving Hamiltonian describes the coherent excitation of an individual atom
between two energies levels, :math:`|a\rangle` and :math:`|b\rangle`, :math:`E_b > E_a`, with
Rabi frequency :math:`\Omega(t)`, detuning :math:`\delta(t)` and phase :math:`\phi(t)`.
```

**Review by lvignoli — APPROVED** [review]
I in fact had this small comment

File: docs/source/conventions.rst
Comment (HGSilveri) [reply docs/source/conventions.rst:140]: If I add the picture, do you think it's worth it to repeat in the text that the energy of b is higher than a? I feel like the picture and the warning should be enough

Comment (lvignoli) [reply docs/source/conventions.rst:140]: I'd rather yes, I add to re-read three time to make sure about the convention between an and b. The warning block had the unintended effect that, being a block of its own, I expected to find only _additional_ information, not the required ones to understand the body text 😅
In fact the warning is really here to insist on the ordering despite the vector space representation, so the energy ordering should be stated in the body text I think.

Comment (lvignoli) [reply docs/source/conventions.rst:140]: (Long comment for a nit, just merge <3 )

**Comment by a-corni** [comment]
Nit: There is a reference to conventions page in the notebook about SLM 

> according to Pulser's conventions for XY basis states

Perhaps we could link both pages ?

**Comment by a-corni** [comment]
This looks good to me !

**Comment by HGSilveri** [comment]
> This looks good to me !

I was still waiting for a reply from @madagra, but I guess I can merge and then make modifications later if necessary

**Pull request merged**
